src/axom/mir/EquiZAlgorithm.hpp

```diff
@@ -0,0 +1,1398 @@
+  void createNodeMapAndSlice(conduit::Node &n_newFields,
+                             axom::Array<axom::IndexType> &nodeMap,
+                             axom::Array<axom::IndexType> &nodeSlice) const
+  {
+    namespace bputils = axom::mir::utilities::blueprint;
+    AXOM_ANNOTATE_SCOPE("createNodeMapAndSlice");
+    SLIC_ASSERT(n_newFields.has_child(originalNodesFieldName()));
+    SLIC_ASSERT(n_newFields.has_child(newNodesFieldName()));
+
+    const axom::IndexType numCleanNodes = m_coordsetView.numberOfNodes();
+
+    // These are the original node ids.
+    const conduit::Node &n_output_orig_nodes =
+      n_newFields[originalNodesFieldName() + "/values"];
+    auto numOutputNodes = n_output_orig_nodes.dtype().number_of_elements();
+    auto outputOrigNodesView =
+      bputils::make_array_view<ConnectivityType>(n_output_orig_nodes);
+
+    // __equiz_new_nodes is the int mask field that identifies new nodes created from blending.
+    const conduit::Node &n_new_nodes_values =
+      n_newFields[newNodesFieldName() + "/values"];
+    const auto maskView = bputils::make_array_view<int>(n_new_nodes_values);
+
+    // Count new nodes created from blending.
+    const int allocatorID = axom::execution_space<ExecSpace>::allocatorID();
+    axom::Array<int> maskOffset(numOutputNodes, numOutputNodes, allocatorID);
+    auto maskOffsetsView = maskOffset.view();
+    RAJA::ReduceSum<reduce_policy, int> mask_reduce(0);
+    axom::for_all<ExecSpace>(
+      numOutputNodes,
+      AXOM_LAMBDA(axom::IndexType index) { mask_reduce += maskView[index]; });
+    const auto numNewNodes = mask_reduce.get();
+
+    // Make offsets.
+    axom::exclusive_scan<ExecSpace>(maskView, maskOffsetsView);
+
+    // Make a list of indices that we need to slice out of the node arrays.
+    nodeSlice =
+      axom::Array<axom::IndexType>(numNewNodes, numNewNodes, allocatorID);
+    auto nodeSliceView = nodeSlice.view();
+    axom::for_all<ExecSpace>(
+      numOutputNodes,
+      AXOM_LAMBDA(axom::IndexType index) {
+        if(maskView[index] > 0)
+        {
+          nodeSliceView[maskOffsetsView[index]] = index;
+        }
+      });
+
+    // Make a node map for mapping mixed connectivity into combined node numbering.
+    nodeMap =
+      axom::Array<axom::IndexType>(numOutputNodes, numOutputNodes, allocatorID);
+    auto nodeMapView = nodeMap.view();
+    axom::for_all<ExecSpace>(
+      numOutputNodes,
+      AXOM_LAMBDA(axom::IndexType index) {
+        if(maskView[index] == 0)
+        {
+          nodeMapView[index] =
+            static_cast<axom::IndexType>(outputOrigNodesView[index]);
+        }
+        else
+        {
+          nodeMapView[index] = numCleanNodes + maskOffsetsView[index];
+        }
+      });
+
+    // Remove fields that are no longer needed.
+    n_newFields.remove(originalNodesFieldName());
+    n_newFields.remove(newNodesFieldName());
+  }
+#endif
+
+  /*!
+   * \brief Perform material interface reconstruction on a single domain.
```

Comment:
[Possible copy/paste] Is this the right doxygen comment for this function?

Reply:
Fixed it.